.github/workflows/test.yml.jinja

```diff
@@ -0,0 +1,59 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+
+name: Test
+
+on:
+  workflow_dispatch:
+    inputs:
+      os-variant:
+        default: 'ubuntu-20.04'
+        type: string
+      python-version:
+        default: '3.9'
+        type: string
+      tox-env:
+        default: 'test'
+        type: string
+      pip-recipe:
+        default: 'requirements/ci.txt'
+        type: string
+      coverage-report:
+        default: false
+        type: boolean
+  workflow_call:
+    inputs:
+      os-variant:
+        default: 'ubuntu-20.04'
+        type: string
+      python-version:
+        default: '3.9'
+        type: string
+      tox-env:
+        default: 'test'
+        type: string
+      pip-recipe:
+        default: 'requirements/ci.txt'
+        type: string
+      coverage-report:
+        default: false
+        type: boolean
+
+jobs:
+  test:
+    name: tests ${{ inputs.os-variant }}, py${{ inputs.python-version }}
+    runs-on: ${{ inputs.os-variant }}
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v3
+        with:
+          python-version: ${{ inputs.python-version }}
+      - run: python -m pip install --upgrade pip
```

Comment:
A point about pinning the versions of everything: we are not pinning the version of `pip` itself. Should we?

Reply:
Have you ever seen problems with it?

Reply:
Not so far.